### PR TITLE
Adding additional client-side date validation

### PR DESCRIPTION
**Pull request by dborgen2**
## What’s changing?

Adding client-side date validation for dates that are too old or too far out in the future. By default it's set to 1900 and 2050, but these years are customizable.

## Why?

Closes NAC-1882

## This PR has:

- [x] **JIRA ticket(s)** (_Must be listed in PR request and branch name should follow convention NAC-123-description_)<description>

- [ ] **Security Related Changes** (_Security Tab questions have been answered on JIRA ticket(s)_)

- [x] **Commented source code** (_required on public classes and methods, and elsewhere as appropriate_)

- [ ] **Documented API changes** (_required on public classes and methods, and elsewhere as appropriate_)
    - [ ] External API documentation - OpenApi docs and Generated Docs (e.g. https://github.com/18F/piipan/tree/dev/docs/openapi) 
    - [ ] Internal API documentation (e.g. https://github.com/18F/piipan/tree/dev/metrics/docs/openapi)   

- [ ] **Developer documentation** (_for new build/test/API changes or complex portions of the system_)

- [x] **Automated unit tests** (_to maintain or increase level of code coverage_)

- [ ] **Changes to IAC** (_add any deployment steps that require manual intervention to the draft release notes_)
    - [ ] Have IaC scripts run successfully against an existing deployed environment?
    - [ ] Have IaC scripts run successfully against a newly deployed environment?
    - [ ] Have any new bash scripts been given the appropriate permissions? (e.g. git update-index --chmod=+x $script)
    - [ ] Do the IaC changes require deleting existing resources?

## Code Review Checklist:

- [x] Automated checks all pass? (code builds successfully, tests pass, test coverage meets/exceeds threshold, etc.) 
- [x] Have all requirements and Acceptance Criteria been met?
- **If there are code changes**
    - [ ] Is there no unnecessary complexity? Code is readable, maintainable, etc.
    - [ ] Do the changes follow SOLID principles?
    - [ ] Are the changes robust? Concurrency considered? Performance? Security (e.g. no sql injections)?
    - [ ] Are newly added dependencies needed and appropriate? Are dependency files (e.g. packages.json & packages.lock.json) updated?
    - [ ] For any new projects/solutions, have they been added to code coverage files?
    - [ ] Public classes & methods are documented    
- **If there are API changes**
    - [ ] Does API make sense? Useful and concise?
    - [ ] Are existing API interfaces/contracts intact? i.e. no breaking changes
    - [ ] Are API tests included?
    - [ ] Are API responses appropriate and documented? (200s, 201, 404s, etc.)
    - [ ] Are internals kept private and not leaked into the API?
    - [ ] External API documentation - OpenApi docs and Generated Docs 
    - [ ] Internal API documentation    
- **If there are IAC changes**
    - [ ] IAC Scripts are readable & maintainable
    - [ ] IAC Scripts are documented
    - [ ] IAC scripts pass required checks
    - [ ] IAC scripts were run successfully by author
- **General Logging Questions**
    - [ ] Do new capabilities (or changed capabilities) include logging?
    - [ ] Is PII not logged?
- **General Error Handling Questions**
    - [ ] Has exception handling been added for new capabilities (or changed capabilities)?
    - [ ] Are errors/exceptions properly communicated to end-users? Were there requirements for user facing error messages?
    - [ ] Are errors/exceptions logged?
    - [ ] Is PII kept out of error/exception logging?
- **If there are Developer Documentation Changes**
    - [ ] General Project documentation (under piipan\docs. e.g. iac.md, architecture.md, etc.)
    - [ ] Github documentation additions/changes (Readmes)
- **General automated test questions**
    - [ ] End-to-End Tests provided (if code implements new user story)
    - [ ] Are tests sufficient?
    - [ ] Corner cases considered?
    - [ ] Are performance tests needed?

### CODE REVIEW

File: query-tool/src/Piipan.QueryTool/Pages/Shared/_EmptyLayout.cshtml [hunk 1]
```diff
@@ -7,8 +7,6 @@
     <title>@ViewData["Title"]</title>
     <base href="/" />
     <link rel="stylesheet" href="~/css/styles.css" />
-    <link href="/Piipan.QueryTool.styles.css" rel="stylesheet" asp-append-version="true" />
```
Comment (bgfoshay) [inline query-tool/src/Piipan.QueryTool/Pages/Shared/_EmptyLayout.cshtml:10]: Question: was the css removal intentional?

File: shared/src/Piipan.Shared.API/Validation/UsaDateAttribute.cs [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+        /// pass in the word "today"
+        /// </summary>
+        /// <param name="MinValue">The string version of the date, or "Today". You can also pass in null to make it so there is no min value.</param>
+        /// <param name="MaxValue">The string version of the date, or "Today". You can also pass in null to make it so there is no max value.</param>
+        public UsaDateAttribute(string? MinValue = "01/01/1900", string? MaxValue = "12/31/2050")
+            : base(typeof(DateTime),
+                  MinValue == Today ? DateTime.Now.ToShortDateString() : MinValue ?? DateTime.MinValue.ToShortDateString(),
+                  MaxValue == Today ? DateTime.Now.ToShortDateString() : MaxValue ?? DateTime.MaxValue.ToShortDateString())
+        {
+            ErrorMessage ??= ValidationConstants.DateBetween1900_2050Message;
```
Comment (bgfoshay) [inline shared/src/Piipan.Shared.API/Validation/UsaDateAttribute.cs:20]: Question: Will this ErrorMessage reflect custom Min & Max values? e.g. what if it's constructed with MinValue=12/20/2022 and MaxValue=12/30/2022?

File: query-tool/src/Piipan.QueryTool/Pages/Shared/_Layout.cshtml [hunk 3]
```diff
@@ -11,8 +11,6 @@
     <title>@ViewData["Title"]</title>
     <base href="/" />
     <link rel="stylesheet" href="~/css/styles.css" />
-    <link href="/Piipan.QueryTool.styles.css" rel="stylesheet" asp-append-version="true" />
```
Comment (bgfoshay) [inline query-tool/src/Piipan.QueryTool/Pages/Shared/_Layout.cshtml:14]: Question: similar question about the removal of this css and the js files?

Comment (dborgen2) [reply query-tool/src/Piipan.QueryTool/Pages/Shared/_Layout.cshtml:14]: Yes, this was a miss on the last story when we were dealing with the sign out page. When looking at that story I realized these CSS files are no longer needed on the _Layout pages, since they are injected by _Host.cshtml when the page is a Blazor page. If it's not a Blazor page we don't need any of the component styles

File: query-tool/src/Piipan.QueryTool/Pages/Shared/_EmptyLayout.cshtml (hunk 1 above)
Comment (dborgen2) [reply query-tool/src/Piipan.QueryTool/Pages/Shared/_EmptyLayout.cshtml:10]: Yes, this was a miss on the last story when we were dealing with the sign out page. When looking at that story I realized these CSS files are no longer needed on the _Layout pages, since they are injected by _Host.cshtml when the page is a Blazor page. If it's not a Blazor page we don't need any of the component styles

File: shared/src/Piipan.Shared.API/Validation/UsaDateAttribute.cs (hunk 2 above)
Comment (dborgen2) [reply shared/src/Piipan.Shared.API/Validation/UsaDateAttribute.cs:20]: Hm... Yea, good point. I was assuming the person would override the error message, but we probably could make the error message dynamic. I can update this

File: query-tool/src/Piipan.QueryTool/Pages/Shared/_EmptyLayout.cshtml (hunk 1 above)
Comment (bgfoshay) [reply query-tool/src/Piipan.QueryTool/Pages/Shared/_EmptyLayout.cshtml:10]: 👍

File: shared/src/Piipan.Shared.API/Validation/UsaDateAttribute.cs (hunk 2 above)
Comment (bgfoshay) [reply shared/src/Piipan.Shared.API/Validation/UsaDateAttribute.cs]: Thanks for updating this.

**Review by bgfoshay — APPROVED** [review]
Thanks for making the error message more flexible. The changes look good!
